configuration/klippy/beacon_true_zero_correction.py

```
@@ -0,0 +1,430 @@
+		probable_x = (r.contact_min[0], r.contact_max[0])
+		probable_y = (r.contact_min[1], r.contact_max[1])
+
+		def in_range(r, value):
+			return r[0] <= value <= r[1]
+
+		if not (
+			in_range(probable_x, range_x[0]) and in_range(probable_x, range_x[1]) and
+			in_range(probable_y, range_y[0]) and in_range(probable_y, range_y[1])):
+
+			self.ratos.console_echo(RATOS_TITLE, 'error', f'The required probing region ({span:.1f}x{span:.1f}) would probe outside the configured contact probing area.')
+			raise self.gcmd.error('The required probing region would probe outside the contact probing area')
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)